### ISSUE TITLE
TypedDict NotRequired field set conditionally passed to Unpack reports error

### ISSUE THREAD

**Issue by Gjgarr** [microsoft/pyright#3741]
**Describe the bug**
The error I'm experiencing is when building a TypedDict and conditionally setting the NotRequired field and then passing the object to an Unpack parameter.

Pylance reports a general type issue that the type of `owner` has become `object`.

**To Reproduce**
```python
from typing import TypedDict
from typing_extensions import NotRequired, Unpack

class Dog(TypedDict):
    owner: str
    found: NotRequired[int]

def display(**args: Unpack[Dog]):
    print(args)
    

# case1: any combination of below is accepted
args: Dog = {
    "owner": "john",
    # "found": 5,
}
# args["found"] = 5
display(**args)
    

# case2: this complains at **args2
args2: Dog = {
    "owner": "jane",
}
if "some_condition":
    args2["found"] = 5
display(**args2)
"""
Argument of type "object" cannot be assigned to parameter "owner" of type "str" in function "display"
  "object" is incompatible with "str" Pylance reportGeneralTypeIssues
"""
```

**Expected behavior**
No errors reported from either case

**VS Code extension or command-line**
Python 3.10.5
VS Code extension, Pylance v2022.7.40

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is fixed in pyright 1.1.264, which I just published. It will also be included in a future release of pylance.
